cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java

```diff
@@ -66,6 +68,13 @@ public Cocos2dxWebView(Context context, int viewTag) {
 
         this.setWebViewClient(new Cocos2dxWebViewClient());
         this.setWebChromeClient(new WebChromeClient());
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```

Comment:
Wait, cocos2d-x supports Android 2.3+ and it still will support. This is why I'm checking the system version at runtime (`Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT`).

We just need to keep minimumSdkVersion == 10, and use newer SDK to compile the code. In fact Cocos2d-x actually already use android-22 as a target platform, but only in the proj.android-studio projects.
 
Cocos2d-x supports two ways of compiling sources:

- old style with Ant (proj.android directories) and
- new style with Gradle (proj.android-studio directories).

In all proj.android-studio projects, cocos2d-x uses android-22 as targetSdkVersion. In all. Check this listening:
```
$ grep -rn "targetSdkVersion " .    
./cocos/platform/android/libcocos2dx/build/outputs/lint-results.html:136:<span class="lineno">   9</span>         <span class="errorspan">targetSdkVersion 22</span>
./cocos/platform/android/libcocos2dx/build/outputs/lint-results.xml:118:        errorLine1="        targetSdkVersion 22"
./cocos/platform/android/libcocos2dx/build.gradle:9:        targetSdkVersion 22
./cocos/platform/android/libcocos2dx-with-controller/build.gradle:9:        targetSdkVersion 22
./templates/cpp-template-default/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./templates/js-template-default/frameworks/runtime-src/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./templates/lua-template-default/frameworks/runtime-src/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/cpp-empty-test/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/cpp-tests/proj.android-studio/app/build/outputs/lint-results.html:45:<span class="lineno">  10</span>         <span class="errorspan">targetSdkVersion 22</span>
./tests/cpp-tests/proj.android-studio/app/build/outputs/lint-results.xml:16:        errorLine1="        targetSdkVersion 22"
./tests/cpp-tests/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/game-controller-test/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/js-memory-gc-tests/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/js-tests/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/lua-empty-test/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/lua-game-controller-test/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/lua-tests/project/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
./tests/performance-tests/proj.android-studio/app/build.gradle:10:        targetSdkVersion 22
```

What we should do is to set the same target for legacy build process. If we don't fix, because of the omission, cocos2d-x will be tied to android-10 functionality only.

Check it out:

- https://medium.com/google-developers/picking-your-compilesdkversion-minsdkversion-targetsdkversion-a098a0341ebd#.j30m6ahvr
- https://developer.android.com/training/basics/supporting-devices/platforms.html?utm_campaign=adp_series_sdkversion_010616&utm_source=medium&utm_medium=blog#version-codes